### PR TITLE
Added query for search url

### LINKED ISSUE #198: Add Query Parameter to Search Functionality
The current search component does not read the query from url

### PR DESCRIPTION
**Pull request by Yashgupta9330**
Resolves #198 

This pr fixes focus on search and type of results and added query for search
Added search in chapter,committee page
Added test for chapter,project,committee pages

### CODE REVIEW

**Comment by Rajgupta36** [comment]
hiii @Yashgupta9330 , it is already include in pr #192 .

**Comment by Yashgupta9330** [comment]
@Rajgupta36, actually in my previous PR for search, I hadn't added the feature to fetch the query from the search URL. After that, I realized there was an issue with focus and a type problem. @arkid15r told me this 2 days ago and advised me to handle it in a separate PR. I contacted @harsh3dev ,he was unable to add the search to the chapter due to the type issue, so I added this feature myself and also included the test for the committee project chapter pages in this PR

**Comment by Rajgupta36** [comment]
> @Rajgupta36, actually in my previous PR for search, I hadn't added the feature to fetch the query from the search URL. After that, I realized there was an issue with focus and a type problem. @arkid15r told me this 2 days ago and advised me to handle it in a separate PR. I contacted @harsh3dev ,he was unable to add the search to the chapter due to the type issue, so I added this feature myself and also included the test for the committee project chapter pages in this PR

yes i created pagination component yesterday but when i integrate it with search bar . I also facing same issue because of statement management . thanks for the pr. i am working on pagination so maybe we require some changes in it. thanks

**Comment by Rajgupta36** [comment]
The problem with this approach is that when we implement pagination, it calls the components multiple times that's make it inefficient. Can we discuss this so that we can create a better pagination system with the better search component?

File: frontend/src/pages/Projects.tsx [hunk 1]
```diff
@@ -11,19 +12,36 @@ import { API_URL } from '../utils/credentials.ts'
...
   const [defaultProjects, setDefaultProjects] = useState<ProjectDataType | null>(null)
+  const [initialQuery, setInitialQuery] = useState<string>('')
 
   useEffect(() => {
     document.title = 'OWASP Projects'
+    const queryParams = new URLSearchParams(window.location.search)
+    const urlQuery = queryParams.get('q')
+    if (urlQuery) {
+      setInitialQuery(urlQuery)
+    }
```
Comment (arkid15r) [inline frontend/src/pages/Projects.tsx]: Isn't this code the same for all pages? Can it be done inside `Search` component?

File: frontend/src/pages/Projects.tsx [hunk 2]
```diff
@@ -11,19 +12,36 @@ import { API_URL } from '../utils/credentials.ts'
...
-        const data = await response.json()
+        let response
+        if (urlQuery) {
+          response = await axios.get(`${API_URL}/owasp/search/project`, {
+            params: { q: urlQuery },
+          })
+        } else {
+          response = await fetch(`${API_URL}/owasp/search/project`)
+          response = await response.json()
+        }
```
Comment (arkid15r) [inline frontend/src/pages/Projects.tsx]: Could you narrow it down to `params` difference only based on `urlQuery`. 

BTW, why do you use both `axios` and `fetch`?

File: frontend/src/components/Search.tsx
Comment (arkid15r) [inline frontend/src/components/Search.tsx:1]: It doesn't show the default search results if I search for something and then make the search query blank.

**Review by arkid15r — CHANGES_REQUESTED** [review]
Please look into these when you get a chance:

File: frontend/src/pages/Projects.tsx (hunk 1 above)
Comment (Yashgupta9330) [reply frontend/src/pages/Projects.tsx]: I have made a hook for this

File: frontend/src/pages/Projects.tsx (hunk 2 above)
Comment (Yashgupta9330) [reply frontend/src/pages/Projects.tsx]: i have used axios in both one api is for without query and another one is with query

File: frontend/src/components/Search.tsx
Comment (Yashgupta9330) [reply frontend/src/components/Search.tsx:1]: I have fixed it

**Comment by Yashgupta9330** [comment]
> The problem with this approach is that when we implement pagination, it calls the components multiple times that's make it inefficient. Can we discuss this so that we can create a better pagination system with the better search component?

I will ping you in slack

**Review by arkid15r — COMMENTED** [review]
https://github.com/OWASP/Nest/pull/192 has been just merged and it has totally new structure. I suggest focusing on adding `?q=` functionality in the new PR instead of fixing the conflicts (up to you).

**Pull request closed without merging**